composer/utils/collect_env.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Helpers to gather system information for debugging and bug reporting.
+
+Leverages PyTorch's :mod:`torch.utils.collect_env` package to gather pertinent system information.
+The following information is additionally collected to faciliate Comopser specific debug:
+
+* Composer version
+* Number of nodes
+* Host processor model name
+* Host processor physical core count
+* Number of accelerators per node
+* Accelerator model name
+
+This package can be invoked as a standalone console script or can be invoked from within
+an application to gather and generate a system environment report.
+
+The module can be invoked by using the entrypoint alias:
+
+.. code-block::
+
+    composer_collect_env
```

Comment:
nit: to denote it's a shell script

```suggestion
    $ composer_collect_env
```